### PR TITLE
Add weighting to extractors

### PR DESCRIPTION
**Pull request by EgorBu**
* propagation of OPTS from base class (`BagsExtractor`) to children classes
* modify TF-IDF formula to utilize `scale` - importance for specific extractor

### CODE REVIEW

File: sourced/ml/repo2/wmhash.py [hunk 1]
```diff
@@ -189,6 +203,7 @@ def uast_to_bag(self, uast):
 class LiteralsBagExtractor(BagsExtractor):
     NAME = "lit"
     NAMESPACE = "l."
+    OPTS = deepcopy(BagsExtractor.OPTS)
```
Comment (vmarkovtsev) [inline sourced/ml/repo2/wmhash.py]: `BagsExtractor.OPTS.copy()`

File: sourced/ml/repo2/wmhash.py [hunk 2]
```diff
@@ -166,11 +174,17 @@ def get_names_from_kwargs(f):
...
+    return dict([(k, v) for k, v in kwargs.items() if k in func_param])
+
+
 @register_extractor
 class IdentifiersBagExtractor(BagsExtractor):
     NAME = "id"
     NAMESPACE = "i."
-    OPTS = {"split-stem": False}
+    OPTS = deepcopy(BagsExtractor.OPTS)
+    OPTS.update({"split-stem": False})
```
Comment (vmarkovtsev) [inline sourced/ml/repo2/wmhash.py]:
```
OPTS = {"split-stem": False}
OPTS.update(BagsExtractor.OPTS)
```

File: sourced/ml/repo2/wmhash.py [hunk 3]
```diff
@@ -209,12 +224,14 @@ def uast_to_bag(self, uast):
 class UastSeqBagExtractor(BagsExtractor):
     NAME = "uast2seq"
     NAMESPACE = "s."
-    OPTS = dict(get_names_from_kwargs(UastSeq2Bag.__init__))
+    OPTS = deepcopy(BagsExtractor.OPTS)
```
Comment (vmarkovtsev) [inline sourced/ml/repo2/wmhash.py]:
```
OPTS = dict(get_names_from_kwargs(UastSeq2Bag.__init__))
OPTS.update(BagsExtractor.OPTS)
```

File: sourced/ml/repo2/wmhash.py [hunk 4]
```diff
@@ -224,12 +241,14 @@ def uast_to_bag(self, uast):
 class UastRandomWalkBagExtractor(BagsExtractor):
     NAME = "node2vec"
     NAMESPACE = "r."
-    OPTS = dict(get_names_from_kwargs(UastRandomWalk2Bag.__init__))
+    OPTS = deepcopy(BagsExtractor.OPTS)
+    OPTS.update(get_names_from_kwargs(UastRandomWalk2Bag.__init__))
```
Comment (vmarkovtsev) [inline sourced/ml/repo2/wmhash.py]: same here

File: sourced/ml/repo2/wmhash.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import codecs
 from collections import namedtuple
+from copy import deepcopy
```
Comment (vmarkovtsev) [inline sourced/ml/repo2/wmhash.py]: rm

**Comment by vmarkovtsev** [comment]
@EgorBu Please discuss with @zurk how to port this to `refactor`

**Comment by zurk** [comment]
@vmarkovtsev I merge it in https://github.com/src-d/ml/pull/139

**Pull request merged**